packages/remix-dev/cli/migrate/migrations/replace-remix-imports/index.ts

```diff
@@ -4,41 +4,116 @@ import { join } from "path";
 import type { PackageJson } from "type-fest";
 
 import * as jscodeshift from "../../jscodeshift";
-import { cleanupPackageJson } from "./cleanupPackageJson";
-import { getTransformOptions } from "./getTransformOptions";
+import {
+  findRemixDependencies,
+  getTransformOptions,
+} from "./getTransformOptions";
 import type { Options } from "./transform";
+import { isAdapter } from "./transform";
+import { isRuntime } from "./transform";
 import type { MigrationFunction } from "../../types";
 import { readConfig } from "../../../../config";
+import { hint } from "../../../../logging";
 
 const transformPath = join(__dirname, "transform");
 
+function* getTasks({
+  packageJson,
+  runtime,
+  adapter,
+}: {
+  packageJson: PackageJson;
+  runtime: string;
+  adapter?: string;
+}): Generator<string> {
+  let remixDeps = findRemixDependencies(packageJson.dependencies);
+
+  // runtime not in deps
+  if (!remixDeps.includes(runtime)) {
+    yield `Add \`@remix-run/${runtime}\` to your dependencies`;
+  }
+
+  // other runtimes in deps
+  let otherRuntimes = remixDeps
+    .filter(isRuntime)
+    .filter((dep) => dep !== runtime);
+  if (otherRuntimes.length > 0) {
+    yield `Remove all unused runtimes from your dependencies: (${otherRuntimes
+      .map((r) => "@remix-run/" + r)
+      .join(",")})`;
+  }
+
+  // adapter not in deps
+  if (adapter && !remixDeps.includes(adapter)) {
+    yield `Add \`@remix-run/${adapter}\` to your dependencies`;
+  }
+
+  // other adapters in deps
+  let otherAdapters = remixDeps
+    .filter(isAdapter)
+    .filter((dep) => dep !== adapter);
+  if (otherAdapters.length > 0) {
+    yield `Remove all unused adapters from your dependencies: (${otherRuntimes
+      .map((r) => "@remix-run/" + r)
+      .join(",")})`;
+  }
+
+  // remix in deps
+  if (Object.keys(packageJson.dependencies || {}).includes("remix")) {
+    yield "Remove `remix` from your dependencies";
```

Comment:
I liked that fact that it was done automatically tbh.

Since we're not doing that anymore and are actually only changing code now, do we still want to keep the `migrate` name? 🤔
The main reason to rename `codemod` to `migration` was that we were doing more than only changing actual code.

Reply:
The issue was that removing `remix` from `package.json` did not remove it from the user's lockfiles. And we can't know what package manager each user is using, so its better to ask them to do it.

As for the naming: I still think `migrate`/`migration` is good as its like a wizard/assistant guiding users through a full migration.

Reply:
We do have some places in our CLI that automatically is installing some things though, which is using `npm`.

#2562 is trying to auto-detect it and run the corresponding command accordingly.
We could maybe use that in here as well in a later phase.

Reply:
The `create` doesn't assume `npm`, but rather asks the user if `npm install` should be run. The user can say "No" and then use whatever package manager they want.

But yes, if we have a stable, reliable way to detect package manager in the future, we can revisit this.

Reply:
We found [@npmcli/package-json](https://www.npmjs.com/package/@npmcli/package-json) as a reliable way to automatically modify the user's `package.json` files and also determined that all major package managers will update their lockfiles from changes in `package.json` when reinstalling.

So now we are back to automatically changing Remix dependencies and removing `remix setup` from `postinstall!